### PR TITLE
feat(docs): Add dynamic navigation (based on mdx content)

### PR DESCRIPTION
**Pull request by sebald**
- Use `gatsby-image` for Logo
- Combine Gatsby's and Marigold's link component
- Create a dynamic nav based on files inside `docs/content`
- Add initial styling for nav
- Move theme switch provider into root wrapper

### CODE REVIEW

File: docs/src/components/Navigation/useNavigation.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
+import { graphql, useStaticQuery } from 'gatsby';
+
```
Comment (ti10le) [inline docs/src/components/Navigation/useNavigation.ts:2]: Very helpful comments for me :)

File: docs/src/components/Navigation/Navigation.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
```
Comment (ti10le) [inline docs/src/components/Navigation/Navigation.tsx:1]: Components with a big letter like MenuItem and ValidationMessage are broken.
You make /menuItem in the nav link but the original is /menu-item

**Review by ti10le — CHANGES_REQUESTED** [review]

File: docs/src/components/Navigation/Navigation.tsx (hunk 2 above)
Comment (sebald) [reply docs/src/components/Navigation/Navigation.tsx:1]: fixed ... freaking gatsby

File: docs/src/components/Navigation/useNavigation.ts (hunk 1 above)
Comment (sebald) [reply docs/src/components/Navigation/useNavigation.ts:2]: thx ^^

**Comment by ti10le** [comment]
@sebald renaming to menuI-iem.mdx is not good, right?

**Comment by sebald** [comment]
> @sebald renaming to menuI-iem.mdx is not good, right?

Well, it will work, but the url will look funny 😬

**Review by ti10le — APPROVED** [review]

**Pull request merged**